### PR TITLE
Update Release to build arm64 image

### PR DESCRIPTION
**Pull request by xcskier56**
This was a quick PR that I'm very positive will NOT work but should be a starting point for discussion. @cainlevy please let me know of any updates I should make.

Thanks!

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -167,3 +203,4 @@ jobs:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
           tags: latest,${{ steps.version.outputs.number }}
+          platform: linux/amd64,linux/arm64
```
Comment (cainlevy) [inline .github/workflows/release.yml:206]: It looks like the Dockerfile [only adds the authn-linux64 binary](https://github.com/keratin/authn-server/blob/main/Dockerfile#L6). You may need to make both binaries available in the build context and then somehow use [`ARG TARGETPLATFORM`](https://docs.docker.com/build/building/multi-platform/#building-multi-platform-images) to add the correct one?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -32,6 +32,32 @@ jobs:
...
+        env:
+          VERSION: ${{ github.ref }}
+          GOOS: linux
+          GOARCH: arm64
+          CGO_ENABLED: 1
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: authn-linux64arm
+          path: dist/authn-linux64arm
```
Comment (cainlevy) [inline .github/workflows/release.yml:60]: you might be able to verify the `linux64Arm` step by running it in your fork, grabbing the artifact, and making sure it executes as expected

**Comment by xcskier56** [comment]
@cainlevy, I'm going to close this PR due to the complexity of automating an arm64 build on Github's amd based runners. Basically what I've found is that I'll have to use Qemu + docker buildx to do the actual build. On my intel mac, this takes nearly 10 minutes. Then to fit into the current build pipeline without major refactoring, you would have to extract the built authn binary from the image and upload as an artifact. 

One route that I found that would help built time is using buildJet to get an ARM runner, but this is a paid service. 

To work around this, I'm going to build our keratin images by hand and store in our own ECR, and when we need to update I'll update the image. 

If/when Github gets their act together and adds ARM runners, I might reopen and finish the implementation.

Thanks,

**Comment by cainlevy** [comment]
Understood! GitHub really does need better runners.

**Comment by AlexCuse** [comment]
@xcskier56 have you looked much at what the cost would be for the BuildJet runners?  I have enough interest in ARM image availability I might be willing to sponsor, especially if it's pure pay-per-minute since we don't release that often.  Self hosting ECR is enough of a pain I think offering an official ARM image could be a worthwhile addition.

**Comment by AlexCuse** [comment]
If it's that cheap I may look into getting signed up over next few weeks.  I could probably use their runners for some other projects too.  There aren't a ton of weaknesses for GitHub CI but this one makes me pull my hair out.

**Comment by xcskier56** [comment]
It is really cheap and took me like 5 minutes to get it running. 

1. Signup with github
2. Setup billing (or use their $5 free)
3. Change the runner to one of their runners

The place you'll likely run into issues is making sure Go is installed on the machine. We use Ruby installed directly on the runner which took me a bit to get going. However, if you use a docker image for building then this might just work. 

One other thing I ran into is that sometimes Docker hub rate-limits pulls for buildjet. All you have to do is add some credentials so that you're not pulling images anonymously. This is in their documentation

**Pull request closed without merging**